R/html-tweak.R

```diff
@@ -206,12 +206,18 @@ tweak_homepage_html <- function(html,
   # logo
   if (!is.null(logo) && bs_version > 3) {
     # Remove logo if added to h1
-    xml2::xml_remove(
-      xml2::xml_find_first(
-        html,
-        ".//h1/img[contains(@src, 'logo')]"
+    # Simple img
+    logo_node <- xml2::xml_find_first(html, ".//h1/img[contains(@src, 'logo')]")
+    if(length(logo_node) > 0) {
```

Comment:
Are you sure that's correct? I would've thought that `xml2_find_first()` always returns an object of length 1, that is sometimes missing.

Either way, I'd probably structure it as two independent steps:

```R
# Bare image
logo_node <- xml2::xml_find_first(html, ".//h1/img[contains(@src, 'logo')]")
if () xml2::xml_remove(logo_node)

# Image in link
logo_node <- xml2::xml_find_first(html, ".//h1/a/img[contains(@src, 'logo')]")
if () xml2::xml_remove(logo_node)
```


Hmmm, or maybe `xml_find_all` is better because `xml_remove()` applied to an empty nodeset probably does nothing:


```R
# Bare image
xml2::xml_remove(xml2::xml_find_all(html, ".//h1/img[contains(@src, 'logo')]"))

# Image in link
xml2::xml_remove(xml2::xml_find_all(html, ".//h1/a/img[contains(@src, 'logo')]"))
```

Reply:
Ok, I made a new try.